src/mpi4py/MPI/Comm.pyx

```diff
@@ -232,6 +232,76 @@ cdef class Comm:
         if self is __COMM_SELF__:  self.ob_mpi = MPI_COMM_SELF
         if self is __COMM_WORLD__: self.ob_mpi = MPI_COMM_WORLD
 
+    # Process Fault Tolerance
+    # -----------------------
+
+    def Revoke(self) -> None:
+        """
+        Revoke a communicator
+        """
+        with nogil: CHKERR( MPI_Comm_revoke(self.ob_mpi) )
+
+    def Is_revoked(self) -> bool:
+        """
+        Indicate whether the communicator has been revoked
+        """
+        cdef int flag = 0
+        with nogil: CHKERR( MPI_Comm_is_revoked(self.ob_mpi, &flag) )
+        return <bint>flag
+
+    def Get_failed(self) -> Group:
+        """
+        Extract the group of failed and acknowledged processes
+        """
+        cdef Group group = Group.__new__(Group)
+        with nogil: CHKERR( MPI_Comm_get_failed(self.ob_mpi, &group.ob_mpi) )
+        return group
+
+    def Ack_failed(self, int num_to_ack: int = -1) -> int:
```

Comment:
@bosilca I have a question about my implementation of this method.

Would the following code with the current MPIX API
```c
MPIX_Comm_failure_ack(comm);
```
be semantically equivalent to the following code with the MPI-5.0 API?
```c
MPI_Comm_size(comm, &num_to_ack);
MPI_Comm_ack_failed(comm, num_to_ack, &num_acked);
```

Reply:
> Some of the Python exception handling can be further simplified using context managers (`with` statement), but I need to think about it. After looking to some of tutorias in SC, I even envision some utility module (e.g. `mpi4py.util.ulfm`) with reusable patterns that would simplify the development of fault tolerant apps in Python.

Cool idea, I would love to see such an utility module with reusable patterns. Let me know if I can help.

> PS: It is quite rare that hard core MPI developers would get interested in Python+MPI. Out of curiosity, may I ask what drove your contribution?

Maybe I'm not such a hardcore MPI developer. I'm using python regularly (not to say almost daily), and it is a wonderful and versatile tool for quick tasks/prototyping. My contribution here was motivated by the fact that I believe that MPI lacks critical capabilities for outside-academia uses, and part of this is the lack of support for fault management. So, in order to make MPI more palatable to these communities you need to address their needs (fault management) and speak their languages (not C).

Reply:
Yes, these two cases lead to similar results. To confirm your understanding, this use case is well documented in the `Advice to users` for the `MPI_Comm_ack_failed` function.

We can also implement `MPIX_Comm_get_acked` with the new API by keeping track of the most recent `num_acked` returned from `MPI_Comm_ack_failed` and then extracting the subgroup with that size from the group returned by `MPI_Comm_get_failed`.